src/main/java/com/google/sps/servlets/InterviewerFeedbackServlet.java

```diff
@@ -0,0 +1,133 @@
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private PersonDao personDao;
+  private Path emailsPath =
+      Paths.get(
+          System.getProperty("user.home") + "/InterviewMe/src/main/resources/templates/email");
+
+  @Override
+  public void init() {
+    init(new DatastoreScheduledInterviewDao(), new DatastorePersonDao());
+  }
+
+  public void init(ScheduledInterviewDao scheduledInterviewDao, PersonDao personDao) {
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.personDao = personDao;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long scheduledInterviewId = Long.parseLong(request.getParameter("interviewId"));
+    int numberOfQuestions = Integer.parseInt(request.getParameter("questionCount"));
+    HashMap<String, String> answers = new HashMap<String, String>();
+    for (int i = 1; i <= numberOfQuestions; i++) {
+      String template = String.format("{{question_%s}}", i);
+      String param = String.format("question%s", i);
+      answers.put(template, request.getParameter(param));
+    }
+
+    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+    // Since Users returned from the LocalUserService (in tests) do not have userIds, here we set
+    // the userId equal to a hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    if (!interviewExists(scheduledInterviewId)) {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+    ScheduledInterview scheduledInterview = getInterview(scheduledInterviewId);
+    answers.put("{{formatted_date}}", scheduledInterview.getDateString());
+    if (isInterviewee(scheduledInterview, userId)) {
+      try {
+        sendFeedback(getInterviewerEmail(scheduledInterview), answers);
+      } catch (Exception e) {
+        e.printStackTrace();
+        response.sendError(HttpServletResponse.SC_REQUEST_TIMEOUT);
+        return;
+      }
+      response.sendRedirect("/scheduled-interviews.html");
+      return;
+    } else {
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+  }
+
+  private boolean interviewExists(long scheduledInterviewId) {
+    return scheduledInterviewDao.get(scheduledInterviewId).isPresent();
+  }
+
+  private ScheduledInterview getInterview(long scheduledInterviewId) {
+    return scheduledInterviewDao.get(scheduledInterviewId).get();
+  }
+
+  private boolean isInterviewee(ScheduledInterview scheduledInterview, String userId) {
+    return scheduledInterview.intervieweeId().equals(userId);
+  }
+
+  private String getInterviewerEmail(ScheduledInterview scheduledInterview) {
+    return personDao
+        .get(scheduledInterview.interviewerId())
+        .map(Person::email)
+        .orElse("Email not found");
```

Comment:
Returning an invalid email here will cause `sendFeedback` to break.

Return Optional<String> instead and return an error status instead.

Reply:
Done